File: 2958. Length of Longest Subarray With at Most K Frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int left=0;
        int ans=0;

        unordered_map<int,int> umap;

        for(int i=0;i<nums.size();i++)
        {
            umap[nums[i]]++;

            while(umap[nums[i]]>k)
            {
                umap[nums[left]]--;
                left++;
            }

            ans=max(ans,i-left+1);
        }
        return ans;
    }
};

File: 2958. Length of Longest Subarray With at Most K Frequency.py
from typing import List
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        left=0
        ans=0
        umap={}
        for i,num in enumerate(nums):
            umap[num] = umap.get(num, 0) + 1

            while(umap[num]>k):
                umap[nums[left]]-=1
                left+=1
            
            ans=max(ans,i-left+1)
        
        return ans